app/controllers/api/providers_controller.rb

```diff
@@ -52,6 +53,18 @@ def delete_resource(type, id = nil, _data = nil)
 
     private
 
+    def update_custom_attributes(provider, custom_attributes)
```

Comment:
No, I'll change the name to remove confusion, but this method is a wrapper that uses Subcollections::CustomATtributes methods.
It adds some checks to the field type and adds metadata to the custom_attributes before sending it to the regular parsing.